modules/mauth-authenticator-akka-http/src/main/scala/com/mdsol/mauth/akka/http/MAuthDirectives.scala

```diff
@@ -91,13 +110,13 @@ trait MAuthDirectives extends StrictLogging {
   /////////////////////////////////////////////
   //  Utility functions
   /////////////////////////////////////////////
-
-  private def extractAuthHeaderDetail(str: String): Option[AuthHeaderDetail] = {
-    if (str.startsWith("MWS ")) {
-      str.replaceFirst("MWS ", "").split(":").toList match {
+  private def extractAuthHeaderDetail(str: String, mauthVersion: MAuthVersion): Option[AuthHeaderDetail] = {
+    val mauthVersionStr = mauthVersion.getValue + " "
+    if (str.startsWith(mauthVersionStr)) {
```

Comment:
We can revert change for this function because it is only used for v1. 
(The extraction logic is buggy too, because a v2 header value starts with "MWS" too)

You can rename or inline this method - it is private so there's no binary compatibility concern

Reply:
Acturaly, it is checking the value starts with mauthVersion.getValue + " " ("MWS ").
Anyway, I reverted the changes of AuthHeaderDetail, extractMAuthHeader and extractAuthHeaderDetail since V2 doesn't need these. See https://github.com/mdsol/mauth-jvm-clients/pull/92/commits/41ff1585613b4d6b13840f5e0a6624440325b903